### PR TITLE
Fix/week old block exit

### PR DESCRIPTION
**Pull request by johba37**
fix for #19

### CODE REVIEW

**Comment by federicobond** [comment]
Hey @johannbarbie, thank you for your contribution! Could you please include the test case that is fixed by your changes, as mentioned in the issue.

**Comment by johba37** [comment]
hi @federicobond, the testcase requires this extra code to simulate progress of time:
```
  function incrementWeekOldBlock() public {
    weekOldBlock = weekOldBlock.add(1);
  }
```
are you sure you want that in there? maybe there is a better way to test `weekOldBlock` variable?

**Comment by federicobond** [comment]
We can make a RootChainMock contract which inherits from RootChain and use that for testing purposes. What do you think?

cc/ @jdkanani

**Comment by jdkanani** [comment]
Yeah. `RootChainMock` would be okay.

**Comment by johba37** [comment]
@federicobond @jdkanani added Mock. have a look. I'm still not happy with that test, because it tests an error which is not there any more.

**Comment by johba37** [comment]
btw, credits for finding this bug go to @troggy, who explained me the code in detail, and added really nice comments: https://github.com/parsec-labs/plasma-contracts/blob/minimum-viable-plasma/contracts/PlasmaRoot.sol#L122-L135

**Review by jdkanani — APPROVED** [review]
Looks good to me.

File: test/RootChain.js [hunk 1]
```diff
@@ -232,6 +232,92 @@ contract('Root chain', function(accounts) {
       assert.equal(user, owner)
       assert.equal(amount.toString(), value)
       assert.deepEqual(pos, posResult.map(p => p.toNumber()))
+    });
+
+    // There is a spectial condition in the MVP spec:
+    // "However, if when calling exit, the block that the UTXO was created in is more than 7 days old, then the blknum of the oldest Plasma block that is less than 7 days old is used instead."
+    // This condition is needed so that very old UTXOs don't overwite the tip of teh priority queue.
```
Comment (federicobond) [inline test/RootChain.js:239]: typo: teh

**Comment by federicobond** [comment]
Merged. Thank you @johannbarbie!

**Pull request merged**